### PR TITLE
Initial commit with some functionality

### PR DESCRIPTION
**Pull request by paulnorthup**
Adds functionality for printing the board, alternating turns, allowing input, error handling on input

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -6,5 +6,9 @@
 """
 
 board = [[None, None, None], [None, None, None], [None, None, None]]
-for i in range(len(board)):
-    print(board[i])
\ No newline at end of file
+
+def print_board(board):
+    for i in range(3):
```
Comment (paulnorthup) [inline app.py]: 3 is a fixed int here, could the loop just iterate through top level objects and be a bit more generic? Should it? if it works this way then maybe that's good enough for now ¯\\_(ツ)_/¯

Comment (paulnorthup) [reply app.py]: vars are good at storing fixed values

Comment (paulnorthup) [reply app.py]: In
* Make it work
* Make it good
* Make it fast (performant)

It's good to qualify "Yes"

File: app.py [hunk 2]
```diff
@@ -10,3 +10,7 @@
 def print_board(board):
     for i in range(3):
         print(board[i])
+
+def get_move():
+    position = input("Enter the position for your next move in '#row# #column#' format\n")
```
Comment (paulnorthup) [inline app.py]: Do we start with 0 or 1 here when counting?

File: test_app.py [hunk 3]
```diff
@@ -17,3 +17,6 @@ def test_updated_position():
     initial = board[position[0]][position[1]]
     updated = board[position[0]][position[1]]
     assert initial != updated
+
+# def test_get_move_len():
```
Comment (paulnorthup) [inline test_app.py:40]: No comments :)

File: app.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 """
 A one-player Python tic-tac-toe game.
 """
-
+mark = "O" # first time alternate_turn is called, mark is changed to "X"
```
Comment (paulnorthup) [inline app.py]: we probably don't need to know this here (comment) 🧀

File: app.py [hunk 5]
```diff
@@ -13,4 +13,19 @@ def print_board(board):
 
 def get_move():
     position = input("Enter the position for your next move in '#row# #column#' format\n")
-    return list(map(int, list(position.split())))
\ No newline at end of file
+    return list(map(int, list(position.split())))
+
+def alternate_turn():
```
Comment (paulnorthup) [inline app.py:40]: probably worth looking for a simple way to alternatrnate/flip a bit in python

Comment (paulnorthup) [reply app.py:40]: It seems like `mark` here is serving multiple purposes, you might be conflating the binary `1` or `0` for your logic with the `X` and `O` for your presentation. We should decouple these

Comment (paulnorthup) [reply app.py:40]:
```
turn = 0

def get_turn(turn_counter):
  # am i on one turn or the other, there are only two
  is_even = turn_counter % 2
  return EVEN if is_even else ODD
  # change to the other turn
  global turn = global turn++ (this is not python.... 🐍 )
```

Comment (paulnorthup) [reply app.py:40]: and also maybe increment_turn is its own thing

File: app.py [hunk 6]
```diff
@@ -13,4 +13,19 @@ def print_board(board):
...
+    global mark
+    if mark == "O":
```
Comment (paulnorthup) [inline app.py]: When you're working with a binary idea you should try to follow binary patterns. Here again we're flipping the bit to represent the icon state, which is separate from the icon itself.

Might make sense to do the binary operation against a 1/0 and have a map for the icon:

```
{0: "X", 1:"O"}
```

File: app.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+def play_move():
+    global board
+
+    position = get_move()
+    alternate_turn()
+    board[position[0]][position[1]] = mark
+    print_board(board)
+
+# NTS: would it be better to create a list of moves already played instead of checking the actual board?
```
Comment (paulnorthup) [inline app.py]: Make the Pull Request and add these as comments

Comment (paulnorthup) [reply app.py]: Yea probably, but that feels heavy compared to where we are today. It's probably worth being reactive for now

File: app.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    elif mark =="X":
+        mark = "O"
+
+
+def play_move():
+    global board
+
+    position = get_move()
+    alternate_turn()
+    board[position[0]][position[1]] = mark
```
Comment (paulnorthup) [inline app.py]: This idea of updating a board location can be wrapped up in its own function

File: app.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    print_board(board)
+
+# NTS: would it be better to create a list of moves already played instead of checking the actual board?
+def check_move(position):
```
Comment (paulnorthup) [inline app.py]: Probably good to use a "handler" here in a general sense

```
if spot is taken
  pass error
  error_handler(spot_taken)
```

do the processing in the handler for whatever issue or error you're raising

**Review by paulnorthup — COMMENTED** [review]
Looks great for a first pass! A few uncle bobs, a couple more discrete functions, and some consistent handling of multiple "binary flip"s should get us in a good position for a v1

File: app.py (hunk 8 above)
Comment (Coniferish) [reply app.py]: In that case should play_move() simply contain calls to the other functions?

Comment (paulnorthup) [reply app.py]: Yup!

File: app.py (hunk 9 above)
Comment (Coniferish) [reply app.py]: I'm not clear on what you're saying I should do here

Comment (paulnorthup) [reply app.py]: I am probably prematurely optimizing here for managing error states. Fine to leave this as is

**Review by Coniferish — COMMENTED** [review]
Should I avoid this pattern of using global variables and side effects to update those variables? See last commit (https://github.com/Coniferish/tictactoe/pull/2/commits/4832e7edbc3b95b22a903f505dfdecfed6562bb1)

**Pull request merged**